File: app/src/main/java/pl/karollisiewicz/shopping/ui/ShoppingActivity.java
package pl.karollisiewicz.shopping.ui;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.navigation.NavController;
import androidx.navigation.ui.AppBarConfiguration;

import pl.karollisiewicz.shopping.R;

import static androidx.navigation.ActivityKt.findNavController;
import static androidx.navigation.ui.ActivityKt.setupActionBarWithNavController;
import static androidx.navigation.ui.NavControllerKt.navigateUp;

public class ShoppingActivity extends AppCompatActivity {
    private AppBarConfiguration appBarConfiguration;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_shopping);
        setSupportActionBar(this.<Toolbar>findViewById(R.id.toolbar));
        setupAppBarConfiguration();
        setupActionBarWithNavController(this,
                findNavController(this, R.id.navHostFragment),
                appBarConfiguration
        );
    }

    private void setupAppBarConfiguration() {
        appBarConfiguration = new AppBarConfiguration.Builder(R.id.shoppingListFragment).build();
    }

    @Override
    public boolean onSupportNavigateUp() {
        final NavController navController = findNavController(this, R.id.navHostFragment);
        return navigateUp(navController, appBarConfiguration) || super.onSupportNavigateUp();
    }
}

File: app/src/main/java/pl/karollisiewicz/shopping/ui/ShoppingActivity.kt
package pl.karollisiewicz.shopping.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import kotlinx.android.synthetic.main.activity_shopping.*
import pl.karollisiewicz.shopping.R

class ShoppingActivity : AppCompatActivity() {
    private lateinit var appBarConfiguration: AppBarConfiguration

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_shopping)
        setSupportActionBar(toolbar)
        setupAppBarConfiguration()
        setupActionBarWithNavController(
            findNavController(R.id.navHostFragment),
            appBarConfiguration
        )
    }

    private fun setupAppBarConfiguration() {
        appBarConfiguration = AppBarConfiguration.Builder(R.id.shoppingListFragment).build()
    }

    override fun onSupportNavigateUp(): Boolean =
        findNavController(R.id.navHostFragment).navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
}
